### ISSUE TITLE
Bug: [no-invalid-void-type] does not allow void in generics when calling a function

### ISSUE THREAD

**Issue by undsoft** [typescript-eslint/typescript-eslint#11997]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=CYUwxgNghgTiAEAzArgOzAFwJYHtXxwAcRUARLKCHAcwB4AVAPgAoBKALngAUYcBbLAGcQDRgG4AUBKIlylGrQBuOLMBatJQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6MAG1IiaNA7RIAGkxQB8DgHcAkkwDiiJgspkAKiUQBBaAHNYAWxP5UGfHERgAvgF1wXxAfIA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare function openDialog<T>(): Promise<T>;

openDialog<void>();
           ~~~~
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-invalid-void-type": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Seems like calling `openDialog<void>()` should be allowed, because it is genuinely useful and not a mistake.

### Actual Result

The rule reports:

> void is only valid as a return type or generic type argument

**Comment by undsoft** [comment]
I'm somewhat confused by this one, because there is a test case that considers `functionGeneric<void>(undefined);` to be invalid code.

At the same time the rule documentation states:

> The following patterns are considered warnings with { allowInGenericTypeArguments: false }:
> logAndReturn\<void\>(undefined);

This implies that with the default setting of `true`, this code should be considered valid by the rule.

**Comment by kirkwaiblinger** [comment]
Hi @undsoft! This is definitely a valid complaint with this rule. 

There have been a number of related issues with no-invalid-void-type that have been aggregated into https://github.com/typescript-eslint/typescript-eslint/issues/8113. In particular, I'd say this issue is an exact duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/6547#issuecomment-1571865881. Therefore, I think this can be closed as a duplicate of the overarching #8113 concerns.
